### DISCUSSION TITLE
HID Class Device with out endpoint question

### DISCUSSION THREAD

**Discussion by KWolfe81** [hathach/tinyusb#954] [Q&A] [answered]
Looking at hid_device.c, lines 382 through 386 are reproduced below.  I'm curious why the report ID is always set to 0 and the type always set to INVALID when the function call seems to be working perfectly and the report ID is right there in the payload...  Am I interpreting/using this incorrectly?

```
  else if (ep_addr == p_hid->ep_out)
  {
    tud_hid_set_report_cb(instance, 0, HID_REPORT_TYPE_INVALID, p_hid->epout_buf, xferred_bytes);
    TU_ASSERT(usbd_edpt_xfer(rhport, p_hid->ep_out, p_hid->epout_buf, sizeof(p_hid->epout_buf)));
  }
```

**Comment by hathach** [comment, marked as answer]
The device stack currently doesn't have hid parser and couldn't tell the difference in follow case:
- 1st byte as report id 
- no report id is present (all bytes are payload).

Therefore it passed that parsing to application.

**Reply by KWolfe81** [reply]
Ok, that makes sense.  I think I got tripped up by 'HID_REPORT_TYPE_INVALID', as in the specific value in the callback is invalid, not the report itself. Thanks again!
